code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -538,6 +556,7 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 		investigate_log("[key_name(user)] started a transfer into [holding].", INVESTIGATE_ATMOS)
 
 /obj/machinery/portable_atmospherics/canister/process_atmos()
+	air_contents.react(src)
```

Comment:
why are you removing the optimization done to reduce needless react calls?

Reply:
ditto